app.py

```diff
@@ -36,38 +35,65 @@
 with app.app_context():
     app.config.from_pyfile('flask.cfg')
 
-# Add theme loader.
-theme_path = path.join(app.config.get('YODA_THEME_PATH'), app.config.get('YODA_THEME'))
-theme_loader = ChoiceLoader([
-    FileSystemLoader(theme_path),
-    app.jinja_loader,
-])
-app.jinja_loader = theme_loader
 
-
-# Load banner configurations
-def load_banner_config():
-    """Load or initialize banner configurations."""
-    config_file_path = path.join(app.config['APP_SHARED_FOLDER'], 'banner_settings.json')
-    default_config = {'banner_enabled': False}
+def load_admin_config():
+    """Load or initialize admin configurations from config file, writing defaults if no config file exists."""
+    config_file_path = path.join(app.config['APP_SHARED_FOLDER'], 'admin_settings.json')
+    def_config = {
+        'banner': {
+            'banner_enabled': False,
+            'banner_importance': False,
+            'banner_message': ''
+        },
+        'YODA_THEME': app.config.get('YODA_THEME')
+    }
 
     try:
+        # If file doesn't exist, create and write the default configuration
         if not path.exists(config_file_path):
-            return default_config
+            with open(config_file_path, 'w') as file:
+                json.dump(def_config, file)
+            return def_config
 
+        # If the file exists, read and return the configuration
         with open(config_file_path, 'r') as file:
             settings = json.load(file)
+            banner_set = settings.get('banner', def_config['banner'])  # Get banner settings or use default
             return {
-                'banner_enabled': settings.get('banner_enabled', False),
-                'banner_importance': settings.get('banner_importance', False),
-                'banner_message': settings.get('banner_message', '')
+                'banner': {
+                    'banner_enabled': banner_set.get('banner_enabled', def_config['banner']['banner_enabled']),
+                    'banner_importance': banner_set.get('banner_importance', def_config['banner']['banner_importance']),
+                    'banner_message': banner_set.get('banner_message', def_config['banner']['banner_message'])
+                },
+                'YODA_THEME': settings.get('YODA_THEME', def_config['YODA_THEME'])
             }
     except Exception:
-        return default_config
+        print("An unexpected error occurred")
+        return def_config
 
 
 app.config['APP_SHARED_FOLDER'] = '/tmp'
-app.config.update(load_banner_config())
+app.config.update(load_admin_config())
+
+# Prepare theme settings
```

Comment:
I thought about it so I default the theme folder name as the displayed text, in case of an added new theme that is not in the mapping.  e.g,. {"tue":"tue"}. See the code from admin.html :
"
<option value="{{ folder }}" {% if folder == current_theme %}selected{% endif %}> **{{ config.theme_mapping.get(folder, folder) }}**</option>
"

What would happen if we let Ansible manages it? Does it give users flexibility to add new mapping pairs, or anything else?

Reply:
Yes, if we let Ansible manage it, it gives the user more flexibility. Moving the theme mapping to Ansible allows the user to easily customize to a more generic installation, without all the different UU related themes for example. Or a more custom installation as they can add their own theme mappings.

Reply:
I have updated the YODA_THEME_MAPPING variable in ansible playbook, see the new PR created for yoda repository: https://github.com/UtrechtUniversity/yoda/pull/451